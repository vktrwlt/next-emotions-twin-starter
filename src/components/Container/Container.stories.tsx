import React from 'react';
import { Story } from '@storybook/react/types-6-0';
import { Container, ContainerProps } from '@/components/Container/Container';

export default {
  title: 'Container',
  component: Container,
};

const Template: Story<ContainerProps> = (args) => <Container {...args} />;

export const Default = Template.bind({});
Default.args = { variant: 'primary', size: 'md' };
