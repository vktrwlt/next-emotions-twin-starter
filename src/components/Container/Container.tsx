import tw, { styled, css, theme } from 'twin.macro';

export interface ContainerProps {
  children: React.ReactNode;
}

const Box = styled.div`
  ${tw`max-w-screen-xl px-8 mx-auto `}
`;

const Container = (props: ContainerProps) => {
  const { children } = props;

  return <Box>{children}</Box>;
};

export { Container };
