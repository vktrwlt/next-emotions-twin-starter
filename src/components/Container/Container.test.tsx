import { Container } from '@/components/Container/Container';
import { render, fireEvent } from '@/utils/test-utils';

describe('<Container />', () => {
  const { asFragment } = render(<Container>Container</Container>);
  it('should render without crashing', () => {
    expect(asFragment()).toMatchSnapshot();
  });
});
