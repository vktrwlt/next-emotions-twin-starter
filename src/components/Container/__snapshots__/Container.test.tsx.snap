// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Container /> should render without crashing 1`] = `
<DocumentFragment>
  .emotion-0 {
  max-width: 1280px;
  padding-left: 2rem;
  padding-right: 2rem;
  margin-left: auto;
  margin-right: auto;
}

<div
    class="emotion-0 emotion-1"
  >
    Container
  </div>
</DocumentFragment>
`;
